src/app/ipv/middleware.js

```diff
@@ -394,18 +395,26 @@ module.exports = {
       const { context } = req?.session || "";
       const pageId = req.session.currentPage;
 
-      const renderOptions = {
-        pageId,
-        csrfToken: req.csrfToken(),
-        pageErrorState: true,
-        context,
-      };
+      const expectedPageId = req.params?.pageId;
 
-      if (pageRequiresUserDetails(pageId)) {
-        renderOptions.userDetails = await fetchUserDetails(req);
+      if (expectedPageId && expectedPageId !== pageId) {
+        return await handleUnexpectedPage(req, res, expectedPageId);
       }
 
-      if (req.method === "POST" && req.body.journey === undefined) {
+      if (req.method === "POST" && req.body?.journey === undefined) {
+        const renderOptions = {
+          pageId,
+          csrfToken: req.csrfToken(),
+          pageErrorState: true,
+          context,
+        };
+
+        if (pageRequiresUserDetails(pageId)) {
+          renderOptions.userDetails = await fetchUserDetails(req);
+        }
+
+        req.renderOptions = renderOptions;
+
```

Comment:
Not sure it made sense to make a call to the backend to fetch user details and input the correct renderOptions before we have decided whether the radio button is empty, but please shout if I've missed something

Reply:
Yeah, I think this makes sense.

Do we also need to create our own render method which fetches user details when necessary? I think it should basically happen any time we render a page...

Reply:
Hmm we could definitely add our own render function which deals with fetching user details if we need to, but there aren't that many pages where we require user details as it currently stands

Reply:
Sure - I don't feel super strongly, I'd just like to avoid more cases like this :)